### DISCUSSION TITLE
Future Plans

### DISCUSSION THREAD

**Discussion by BadDogSkyrim** [BadDogSkyrim/PyNifly#182] [Ideas]
General thoughts and info about where pyNifly is going:

We're about to go into a period of multiple major-version updates, which won't be backwards compatible. Sorry about that. Not backwards compatible means you can't import with an older version and export with the new version.

Reason for that is that I want to improve the fidelity of the Blender representation. Any shader attribute Blender can handle should be represented. Collisions should be Blender objects with physics, where possible. Constraints should be Blender constraints. Now that there's full support for shader attributes, next step is things like UV scale and offset, environment textures, and so forth.

MOPP collisions still aren't supported. I'd like to support them but it's a can of worms. I'd be more excited about it if I could handle collisions for FO4 at all. I might go look at the tools there and see if I can at least integrate with them.

Animation support exists but I'm not convinced it's robust. I'll be working on making it more complete.

**_As for Starfield_** my original plan was to jump on Starfield ASAP. But there's a lot of activity there--a limited importer already exists--and I'm not sure what value I'd be adding. I may hold off and see how it goes. I plan to do Starfield modding myself--if I don't like the tools I may update pyNifly at that point.

Comments welcome.

**Comment by podvornyakva** [comment]
Starfield is crap. As as Fallout 4. TES - that is a best of Besethda. Even now Skyrim, with mods, looks better than both Fallout 4 and Starfield(you made tool for moders, btw). More important TES have wide and deep LOR. It is Legendary. I'll wait for TES VI... IMHO.
